monix-eval/shared/src/main/scala/monix/eval/internal/TaskConversions.scala

```diff
@@ -106,8 +106,8 @@ private[eval] object TaskConversions {
         val cancelable = SingleAssignCancelable()
         conn push cancelable
 
-        val io = F.runCancelable(fa)(new CreateCallback[A](conn, cb))
-        cancelable := Cancelable.fromIOUnsafe(io.unsafeRunSync())
+        val syncIO = F.runCancelable(fa)(new CreateCallback[A](conn, cb))
+        cancelable := Cancelable.fromIO(F.toIO(syncIO.unsafeRunSync()))
```

Comment:
With this change `Cancelable.fromIOUnsafe` is unused but maybe it should stay here - how can I assess whether I need to guarantee idempotency or not?